### DISCUSSION TITLE
problem with FastClick at high rates and large packet sizes

### DISCUSSION THREAD

**Discussion by hamidgh09** [tbarbette/fastclick#390] [Q&A]
Hi Tom, 

I recently noticed an issue with FastClick when it receives traffic with a high rate (ie >60 Gbps) and actual packet sizes.

In my experiment, I run a simple forwarder on FastClick which does nothing but receiving packets from dpdk and discarding them.
When I increase the number of cores to 8, I expect FastClick to work fine even at 100Gbps, but in practice NIC starts discarding packets at 60Gbps and the throughput drops to <40Gbps. 
Everything is fine With the exact same traffic when packets are truncated to small sizes (eg, 64 or 128 Bytes). 

I have tried with different DPDK configurations and the problem still persists. Also I know NIC is not the problem as a simple DPDK application that I have written with default configurations works fine.

I wonder if there is a specific configuration in FastClick that I am missing or you have encountered the same issue before :p

Thanks in advance :)

Update1: If it helps, FastClick simply reaches 100Gbps with 1 or 2 cores, the issue raises with 4 or more cores.

**Comment by tbarbette** [comment]
Hi Hamid! 

What is your configure line and DPDK version? And maybe an example problematic pipeline?

ConnectX5 ? 

Cheers,
Tom

**Reply by hamidgh09** [reply]
I have tried multiple things. This is the latest configuration I have:

configure line:
`
./configure --enable-dpdk --enable-intel-cpu --verbose --enable-select=poll "CFLAGS=-O3" "CXXFLAGS=-std=c++17 -O3" --disable-dynamic-linking --enable-poll --enable-bound-port-transfer --enable-local --disable-task-stats --enable-cpu-load --enable-dpdk-packet --disable-clone --disable-dpdk-softqueue --enable-research --disable-sloppy --enable-user-timestamp --enable-flow
`
click file:
```
DPDKInfo(NB_MBUF 1048576, MBUF_SIZE 8192, MBUF_CACHE_SIZE 512)
define($nfport 0000:17:00.0)

fd0 :: FromDPDKDevice($nfport, SCALE SHARE, PROMISC true, PAUSE none, MAXTHREADS 8, N_QUEUES 8, NUMA 1, VERBOSE 99, NDESC 1024, BURST 64, RSS_AGGREGATE false) 
// -> ToDPDKDevice($nfport, VERBOSE 99, TIMEOUT -1, BLOCKING false)
-> Discard;

DriverManager(
                read fd0.rss_reta,
                wait,
                read fd0.xstats
)
```

The DPDK versions I tried are v22.07 and v21.11 and the NIC is ConnectX-6

Hamid

P.S. I am using the CAIDA and our campus trace file for sending the traffic, btw.

**Reply by tbarbette** [reply]
Can you share the beginning of the failing CAIDA trace? And maybe the beginning of the non-failing one so we can compare?

**Comment by tbarbette** [comment]
Hi @hamidgh09, did you ever solve this?

**Reply by hamidgh09** [reply]
Hi, I'm sorry for missing your last comment! 

I just know it had something to do with the length value in the headers. 
The problem fixed after rewriting them on both MAC and IP headers!

**Reply by tbarbette** [reply]
Ok thanks, will close and if someone has the same problem we can dig again ^^
